packages/data/src/components/use-select/index.js

```diff
@@ -193,7 +193,11 @@ function useStaticSelect( storeName ) {
 function useMappingSelect( suspense, mapSelect, deps ) {
 	const registry = useRegistry();
 	const isAsync = useAsyncMode();
-	const store = useMemo( () => Store( registry, suspense ), [ registry ] );
+	const store = useMemo(
+		() => Store( registry, suspense ),
+		[ registry, suspense ]
+	);
+	// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
🤔 Why do we run into yet another ESLint warning here? Is it because ESLint thinks `mapSelect` needs to be part of the `deps` too? If that's the case, should we add a comment explaining why this is alright?

Reply:
While technically valid, this pattern confuses ESList React hook rules. Even when you have something like the code below, you still get the following warning - `React Hook useCallback was passed a dependency list that is not an array literal. This means we can't statically verify whether you've passed the correct dependencies.`.

I'll try to come up with a short explanation and include the reason for disabling.

```js
function useFauxSelect( deps ) {
	return useCallback( () => {}, deps );
}
```

Reply:
@tyxla, what do you think about something like this?

```
These are "pass-through" dependencies from the parent hook,
and the parent should catch any hook rule violations.
```

P.S. There's probably a React/technical term for a similar pattern, but I couldn't find one 😅

Reply:
I think this is well described 👍 I can think of other ways to say it, but they are less direct, while your suggestion immediately clicked.

Reply:
Thanks for double-checking!